crates/katana/executor/src/abstraction.rs

```diff
@@ -107,7 +108,8 @@ pub struct ExecutionOutput {
     /// The state updates produced by the executions.
     pub states: StateUpdatesWithDeclaredClasses,
     /// The transactions that have been executed.
-    pub transactions: Vec<(TxWithHash, Option<Receipt>)>,
+    /// TODO: do we want a struct instead?
+    pub transactions: Vec<(TxWithHash, Option<Receipt>, TxExecInfo)>,
```

Comment:
Here we have the receipt being optional. So we may not want to use `TxWithOutcome`. However, a struct may be more expressive as the tuple?

Reply:
seems better if if we can return `Box<dyn TransactionExecutionOutput>` instead? relate to the reason my comment below.

> maybe what we can do is we store it as Box<dyn TransactionExecutionOutput> and only take out the receipt and txinfo after the tx execution phase is done during block production process, instead of doing it after each tx execution.